File: solutions/2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.ts
function findGCD(nums: number[]): number {
  let mn = nums[0], mx = nums[0];
  for (const x of nums) {
    if (x < mn) mn = x;
    if (x > mx) mx = x;
  }
  let a = mn, b = mx;
  while (b !== 0) {
    const t = a % b;
    a = b;
    b = t;
  }
  return a;
}

File: solutions/2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.cpp
class Solution {
public:
    int gcd(int a,int b)
    {
        if(a == 0)
        {
            return b;
        }
        return gcd(b%a,a);
    }
    int findGCD(vector<int>& nums) {
        int a=INT_MAX,b=INT_MIN;
        for(int i=0;i<nums.size();i++)
        {
            a=min(a,nums[i]);
            b=max(b,nums[i]);
        }
        return gcd(a,b);
    }
};